### ISSUE TITLE
RadDataForm: TimePicker 24 hour format

### ISSUE THREAD

**Issue by bori87** [telerik/nativescript-ui-feedback#327]
How to set TimePicker to 24 hour format.
![timer](https://user-images.githubusercontent.com/29092029/31270448-b3350f5e-aa84-11e7-8007-535f44948577.png)

**Comment by tgpetrov** [comment]
Hi @bori87,
The `Time` editor internally does not use the [TimePicker](https://developer.android.com/reference/android/widget/TimePicker.html) that you reference, but the [TimePickerDialog](https://developer.android.com/reference/android/app/TimePickerDialog.html). Its only option to change whether the format is 24H or not is in its [constructor](https://developer.android.com/reference/android/app/TimePickerDialog.html#TimePickerDialog(android.content.Context,%20android.app.TimePickerDialog.OnTimeSetListener,%20int,%20int,%20boolean)). This means that you have no option to change the format later and currently we haven't exposed a way to provide it before the widget is constructed as well. We will consider such feature for future releases, but I can't commit to a certain date when this feature would be available.

**Comment by bori87** [comment]
Thank you but i dont know how to access the TimePickerDialog. There is only .getDialog() method in my timePickerEditor.

**Comment by tgpetrov** [comment]
Yes, the .getDialog() will return the TimePickerDialog that the editor uses. Unfortunately this will not help you as the TimePickerDialog doesn't have an option to change its format. We will have to implement another property for the editor that you would be able to set and change the format. In other words, until we provide a specific option (in a future release), you can't change the format of this editor.

**Comment by bori87** [comment]
This is very bad news for me.  According to https://i.stack.imgur.com/43dwH.jpg most people use 24h format.  Can i use custom <TimePicker> inside RadDataForm or only RadDataForms compononent are allowed?

**Comment by tgpetrov** [comment]
You can create your own custom editor as demonstrated [here](http://docs.telerik.com/devtools/nativescript-ui/Controls/NativeScript/DataForm/Editors/dataform-editors-custom). The article shows how to create an editor with Button-s but you can use the same approach for other native components with the required setup - for example - [UIDatePicker](https://developer.apple.com/documentation/uikit/uidatepicker) for iOS and [TimePickerDialog](https://developer.android.com/reference/android/app/TimePickerDialog.html) (or [TimePicker](https://developer.android.com/reference/android/widget/TimePicker.html)) for Android.

**Comment by atheros** [comment]
And if it was opensource, some very nice guy could do a PR to fix it, someone a bit more lazy could copy the existing editor, rename it, change that constructor and have proper 24h time selector, then share it on stackoverflow or somewhere else.

But no, we chose to use closed-source software, so it's only our fault we end up f***ed. Fun, right?

**Comment by elena-p** [comment]
Hi @bori87, @atheros,

Contributions are now possible after signing a mutual NDA. If you or anyone else is interested to contribute, contact us at nativescriptplugins@progress.com.

**Comment by vytautas-pranskunas-** [comment]
Any progress on this?

**Comment by Klunk75** [comment]
> Any progress on this?

+1
